lib/cocoapods/bazel/target.rb

```diff
@@ -562,7 +562,24 @@ def test_kwargs
 
       def app_kwargs
         # maps to kwargs listed for https://github.com/bazelbuild/rules_apple/blob/master/doc/rules-ios.md#ios_application
-        {
+
+        # If there's a SWIFT_DEPLOYMENT_TARGET version set, use that for the
+        # minimum version. It's not currently supported or desirable in rules_ios to have
+        # these distinct, however xcconfig supports that.
+        swift_deployment_target = resolved_value_by_build_setting('SWIFT_DEPLOYMENT_TARGET')
+
+        llvm_target_triple_os_version = resolved_value_by_build_setting('LLVM_TARGET_TRIPLE_OS_VERSION')
+        if llvm_target_triple_os_version
+          # For clang this is set ios9.0: take everything after the os name
+          version_number = llvm_target_triple_os_version.match(/\d.*/)
+          clang_build_os_version = version_number if version_number
+        end
+
+        puts 'warning: had different os versions' if swift_deployment_target != clang_build_os_version
```

Comment:
Worth raising instead?

Reply:
it's unlikely these are set by a user but it seems fine to do that assertion